core-backend/sandbox/src/funcs.rs

```diff
@@ -883,7 +883,7 @@ where
         };
 
         ctx.err = match f() {
-            Ok(duration) => FuncError::Terminated(TerminationReason::Wait(duration)),
+            Ok(duration) => FuncError::Terminated(TerminationReason::Wait(duration, false)),
```

Comment:
As I mentioned in issue, I prefer here to pass different bool value due to:
if we go with maximum queried by `wait_up_to(max)`, then reincarnation, otherwise go without it

Reply:
The main idea is:
- gr_wait waits as much as it can, drops message afterward as out of gas one
- gr_wait_for(duration) have guarantees to be waken in `duration` **with** gas (requeued)
- gr_wait_up_to(duration) is something between: in case when contract can pay for whole duration it should work as gr_wait_for, otherwise (in case when it has less money than whole duration) similar to gr_wait